### ISSUE TITLE
problem with fstd2nc

### ISSUE THREAD

**Issue by gulilat-diro** [neishm/fstd2nc#22]
Hi,
I'm having a problem with fstd2nc while converting from rpn format to netcdf on ppp3.

here is the error message
......

/space/hall3/sitestore/eccc/cmd/n/gud001/suites_data/cansips/hub/work/20191130000000/c3s_data/cancm4/ens_mem_01>fstd2nc -f --nc-format NETCDF4_CLASSIC SV_allmonths_20191130st_ens_01_sellev.rpn SV_allmonths_20191130st_ens_01_sellev.nc
Found 1 RPN input file(s)
Saving netCDF fileTraceback (most recent call last):
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20190903.0_all/bin/fstd2nc", line 11, in <module>
    load_entry_point('fstd2nc==0.20190903.0', 'console_scripts', 'fstd2nc')()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20190903.0_all/lib/python/fstd2nc/__main__.py", line 107, in _fstd2nc_cmdline_trapped
    _fstd2nc_cmdline (*args, **kwargs)
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20190903.0_all/lib/python/fstd2nc/__main__.py", line 101, in _fstd2nc_cmdline
    buf.to_netcdf(outfile, nc_format=nc_format, global_metadata=global_metadata, zlib=zlib, compression=compression, progress=progress)
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20190903.0_all/lib/python/fstd2nc/mixins/netcdf.py", line 376, in to_netcdf
    for r,shape,v,ind in bar.iter(sorted(io)):
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20190812_ubuntu-18.04-amd64-64/lib/python/progress/__init__.py", line 174, in iter
    with self:
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20190812_ubuntu-18.04-amd64-64/lib/python/progress/__init__.py", line 129, in __enter__
    self.start()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20190812_ubuntu-18.04-amd64-64/lib/python/progress/__init__.py", line 162, in start
    self.update()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20190903.0_all/lib/python/fstd2nc/mixins/__init__.py", line 179, in update
    super(_ProgressBar,self).update()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20190812_ubuntu-18.04-amd64-64/lib/python/progress/bar.py", line 71, in update
    filled_len = self.width * self.progress
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20190812_ubuntu-18.04-amd64-64/lib/python/progress/__init__.py", line 155, in progress
    return min(1, self.index / self.max)
ZeroDivisionError: division by zero
.....
I will be grateful if you could comment on this

**Comment by neishm** [comment]
Hi @gulilat-diro,

Thanks for the bug report.  What happened is that particular file contains a variable called "SV", and the converter thinks anything called "SV" is a vertical coordinate for "series" files:
https://github.com/neishm/fstd2nc/blob/958ce511f07f65bc7a13192f7397de61a043f962/fstd2nc/mixins/series.py#L94
The end result is the "SV" variable doesn't get treated as a normal variable, and is ignored for the conversion.  This results in there being 0 records converted, which causes a division-by-zero for the progress bar logic.

I'll have to modify the code to check what SV actually is (1D coordinate or 2D variable), and handle it accordingly.  This may take some time to work out.

In the mean time, one workaround that might work: zap the SV record with `editfst` and rename to something else (like SVX), then it should convert.  You could then rename it back during the conversion with the option `--rename SVX=SV`.

**Comment by neishm** [comment]
Hi @gulilat-diro,

Another temporary workaround (that avoids `editfst`) is to create a local script that contains the following lines:
```python
#!/usr/bin/env python

# Force SV to be treated as a normal variable (not a series coordinate).
from fstd2nc.mixins.series import Series
def _makevars (self):
  self._meta_records = tuple(r for r in self._meta_records if r != 'SV')
  super(Series,self)._makevars()
Series._makevars = _makevars

# Continue with fstd2nc conversion.
from fstd2nc.__main__ import _fstd2nc_cmdline_trapped
_fstd2nc_cmdline_trapped()
```
If you run the conversion through that script, it should avoid the problems with SV.

Hope that helps,
Mike

**Comment by gulilat-diro** [comment]
Hi Mike,
Thanks for the explanation. I used your first option for the workaround i.e. r.diag  newnam to change the variable name and it works now.

cheers,
Gulilat

**Comment by neishm** [comment]
Hi @gulilat-diro,

This problem should now be fixed in the latest version of the converter (0.20200717.0).

Would it be alright if I included one of these files (SV_allmonths_20191130st_ens_01_sellev.rpn) in the test suite on the science network?  It would be useful for regression testing, in case something breaks again in future versions.

Thanks,
Mike

**Comment by neishm** [comment]
This problem reared in head again in recent versions (0.20211013.0 and later).  It should now be fixed again in latest version (0.20220204.1).
